### DISCUSSION TITLE
How to create Multi-stage Docker image with Kafka-connect?

### DISCUSSION THREAD

**Discussion by imrohankataria** [strimzi/strimzi-kafka-operator#9251] [Q&A]
Hi,

My Team is using Strimzi Kafka connect in enterprise setting, and because of the image straight from quay.io we are seeing lots of vulnerabilities in the image.

To resolve those CVEs, I have been able to drop the folders/utilities that are not being used eg. kafa-exporter, cruise-control, etc

But the issues are still coming up because of Shasum history in the docker image. One resolution I tried before was squashing the images, but it is not working anymore. 

Another approach is creating a Multi-stage image, which Im failing to create. 

Will appreciate your help on this.

Code I have been trying to use -

```
FROM quay.io/strimzi/kafka:0.36.1-kafka-3.5.1 as builder

# Switch to root user to make modifications
USER root

# Remove unwanted components
RUN rm -rf /opt/kafka-exporter /opt/cruise-control

# Remove the old guava library
RUN rm /opt/kafka/libs/guava-30.1-jre.jar

# # Stage 2: Final Alpine-based image
FROM alpine:latest

# # Create Kafka user with similar properties to Strimzi's Kafka user
RUN addgroup -g 1001 -S kafka && adduser -u 1001 -S kafka -G kafka

# # Copy necessary files from the builder stage, retaining permissions
COPY --from=builder --chown=kafka:kafka /opt/kafka /opt/kafka

# # Add full permissions for the 'kafka' group to the /opt/kafka directory
RUN chmod -R g+rwx /opt/kafka

# Switch to Kafka user
USER kafka
```

**Comment by scholzj** [comment]
Why do you start a new discussion instead of continuing in the old one? https://github.com/orgs/strimzi/discussions/8939

**Reply by imrohankataria** [reply]
@scholzj Can I update the title of old one?

**Reply by scholzj** [reply]
Sure, that should be possible. There should be an "Edit" button in the top right corner.

**Reply by imrohankataria** [reply]
Updated - https://github.com/orgs/strimzi/discussions/8939
